src/provable/binable.ts

```diff
@@ -30,6 +30,44 @@ function withVersionNumber<T>(
   };
 }
 
+type Tuple<T> = [T, ...T[]] | [];
+
+function compose<Types extends Tuple<any>>(
```

Comment:
compose two binables to create a binable for the composed type. this is used in this PR to re-create Mina's encoding of public keys in JS

Reply:
nit: Can we call this `tuple` or `pair`? There are a few ways to compose values and the tupling action could be more explicit from the name